jina/clients/mixin.py

```diff
@@ -57,18 +60,31 @@ async def _get_results(*args, **kwargs):
             if return_results:
                 return result
 
-        return run_async(
-            _get_results,
-            inputs=inputs,
-            on_done=on_done,
-            on_error=on_error,
-            on_always=on_always,
-            exec_endpoint=on,
-            target_peapod=target_peapod,
-            parameters=parameters,
-            request_size=request_size,
-            **kwargs,
-        )
+        with ExitStack() as stack:
```

Comment:
Isnt this also done in the regular Client and not only in the Flow context? Can we guarantee that this never is executed if a user just uses the regular Client?

Reply:
Maybe you can use self.client to set up the context somehow? So you dont need to do it every time?

Reply:
there is no point on using a stack when u have only one context manager to handle

Reply:
> isn't this also done in the regular Client and not only in the Flow context? Can we guarantee that this never is executed if a user just uses the regular Client?


Thanks. When just running the client, there is no infrastructure attribute and therefore the normal context is used.

Reply:
> there is no point on using a stack when u have only one context manager to handle

The point is to have more compact code.